policies/model/src/main/java/org/eclipse/ditto/policies/model/signals/commands/query/PolicyCheckPermissionsCommand.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.policies.model.signals.commands.query;
+
+import static org.eclipse.ditto.base.model.common.ConditionChecker.checkNotNull;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.base.model.json.FieldType;
+import org.eclipse.ditto.base.model.json.JsonParsableCommand;
+import org.eclipse.ditto.base.model.json.JsonSchemaVersion;
+import org.eclipse.ditto.base.model.signals.commands.AbstractCommand;
+import org.eclipse.ditto.base.model.signals.commands.CommandJsonDeserializer;
+import org.eclipse.ditto.json.JsonFactory;
+import org.eclipse.ditto.json.JsonField;
+import org.eclipse.ditto.json.JsonFieldDefinition;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonObjectBuilder;
+import org.eclipse.ditto.json.JsonPointer;
+import org.eclipse.ditto.policies.model.PolicyId;
+import org.eclipse.ditto.policies.model.ResourcePermissionFactory;
+import org.eclipse.ditto.policies.model.ResourcePermissions;
+import org.eclipse.ditto.policies.model.signals.commands.PolicyCommand;
+
+@Immutable
+@JsonParsableCommand(typePrefix = PolicyCommand.TYPE_PREFIX, name = PolicyCheckPermissionsCommand.NAME)
+public final class PolicyCheckPermissionsCommand extends AbstractCommand<PolicyCheckPermissionsCommand>
```

Comment:
I would rename it to `CheckPolicyPermissions` - the commands normally don't have `*Command` suffix.
And `Policy` as prefix also seems strange to me.

Maybe even `CheckPermissions` would be sufficient. (Then we would have to rename the `CheckPermissions` in gateway .. but this is only gateway internal anyways).

Javadoc is missing;
Please add unit tests